workspace.go

```diff
@@ -396,6 +411,12 @@ type VCSRepoOptions struct {
 	GHAInstallationID *string `json:"github-app-installation-id,omitempty"`
 }
 
+type WorkspaceSettingOverwritesOptions struct {
+	// using `json` rather than `json-api` so that nil values are truly omitted. in `jsonapi` the json key will still be present
+	ExecutionMode *bool `json:"execution-mode,omitempty"`
+	AgentPool     *bool `json:"agent-pool,omitempty"`
```

Comment:
This whole scheme is extremely indirect, so let's start things off right by establishing some conventions! In the create/update options version of these overwrites struct, let's document each field with the exact identity of the parent setting that gets used if the workspace doesn't set its own value. (We can be lazy in the value version of the struct, because the whole point of the scheme is so that people can just usually just read the top-level settings values and go home instead of doing inheritance math.)

```suggestion

	// If false, the workspace will defer to its organization's DefaultExecutionMode value.
	ExecutionMode *bool `json:"execution-mode,omitempty"`
	// If false, the workspace will defer to its organization's DefaultAgentPool value.
	AgentPool     *bool `json:"agent-pool,omitempty"`
```

(Extra blank line at the top is so nothing interprets your comment about jsonapi field tags as being part of the documentation for the first field.)

Reply:
Love this! Only caveat I might add is that currently we are working on allowing the project to overwrite this value too 😓 So I'll just slightly modify what you've suggested here!